src/utils/gaa-test.js

```diff
@@ -2977,23 +2976,23 @@ describes.realWin('GaaMetering', {}, () => {
     beforeEach(() => {
       GaaMetering.gaaUserPromiseResolve_ = undefined;
     });
-    it('sets up the promise to return the gaaUser', () => {
-      GaaMetering.getGaaUserPromise().then((gaaUser) => {
-        expect(gaaUser).to.equal('test gaaUser');
-      });
-      GaaMetering.setGaaUser('test gaaUser');
+
+    it('sets up the promise to return the gaaUser', async () => {
+      const gaaUser = 'test gaaUser';
+      const gaaUserPromise = GaaMetering.getGaaUserPromise();
+      GaaMetering.setGaaUser(gaaUser);
+      expect(await gaaUserPromise).to.equal(gaaUser);
     });
   });
 
   describe('getLoginPromise', () => {
-    beforeEach(() => {
-      GaaMetering.loginPromiseResolve_ = undefined;
-    });
-    it('sets up the promise', () => {
-      GaaMetering.getLoginPromise().then(() => {
-        expect(GaaMetering.loginPromiseResolve_).to.have.been.called();
-      });
+    beforeEach(() => {});
+    it('sets up the promise', async () => {
+      const promise = GaaMetering.getLoginPromise();
+      sandbox.spy(GaaMetering, 'loginPromiseResolve_');
       GaaMetering.resolveLogin();
+      await promise;
+      expect(GaaMetering.loginPromiseResolve_).to.have.been.called;
```

Comment:
I think this should just expect `getLoginPromise.to.be.fulfilled`, rather than looking at the private field `loginPromiseResolve_` (existing issue, fine if you don't want to change it here).

Reply:
Updated 👍